geoips/compare_outputs.py

```diff
@@ -214,95 +214,76 @@ def images_match(output_product, compare_product, fuzz="5%"):
-        fuzz,
-        output_product,
-        compare_product,
-        out_diffimg,
-    ]
-
-    exact_call_list = [
-        "compare",
-        "-verbose",
-        "-quiet",
-        "-metric",
-        "ae",
-        output_product,
-        compare_product,
-        exact_out_diffimg,
-    ]
-
-    LOG.info("**Running %s", " ".join(call_list))
-    fullimg_ret = subprocess.run(
-        call_list, stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+    # Determine the number of pixels that are mismatched
+    num_pix_mismatched = pixelmatch(
+        out_img, comp_img, diff_img, includeAA=True, alpha=0.33, threshold=0.05
     )
-    LOG.debug(fullimg_ret.stdout)
-
+    # Currently, return 0 ONLY if the images are exactly matched.  Eventually
+    # we may update this to allow returning 0 for "close enough" matches.
+    if np.all(diff_arr == 0) and num_pix_mismatched == 0:
+        fullimg_retval = 0
+    else:
+        fullimg_retval = 1
+    # Write out the exact image difference.
+    LOG.info("**Saving exact difference image")
+    diff_img.save(exact_out_diffimg)
     LOG.info("**Done running compare")
 
-    # call_list = ['compare', '-verbose', '-quiet',
-    #              '-metric', 'rmse',
-    #              '-dissimilarity-threshold', '{0:0.15f}'.format(threshold),
-    #              '-subimage-search',
-    #              output_product,
-    #              compare_product,
-    #              out_diffimg]
-
-    # LOG.info('Running %s', ' '.join(call_list))
-
-    # subimg_retval = subprocess.call(call_list)
-    # if subimg_retval != 0 and fullimg_retval != 0:
-    #     call_list = ['compare', '-verbose', '-quiet',
-    #                  '-metric', 'rmse',
-    #                  '-subimage-search',
-    #                  output_product,
-    #                  compare_product,
-    #                  out_diffimg]
-    #     subprocess.call(call_list)
-    #     LOG.info('    ***************************************')
-    #     LOG.info('    *** BAD Images do NOT match exactly ***')
-    #     LOG.info('    ***************************************')
-    #     return False
-    if fullimg_ret.returncode != 0:
+    # If the images do not match exactly, print the output image, comparison image,
+    # and exact diff image to log, for easy viewing.  Return False.
+    if fullimg_retval != 0:
         LOG.interactive("    ***************************************")
         LOG.interactive("    *** BAD Images do NOT match exactly ***")
         LOG.interactive("    ***   output_product: %s ***", output_product)
         LOG.interactive("    ***   compare_product: %s ***", compare_product)
-        LOG.interactive("    ***   out_diffimg: %s ***", out_diffimg)
-        LOG.interactive("    ***   exact_out_diffimg: %s ***", exact_out_diffimg)
+        LOG.interactive("    ***   exact dif image: %s ***", exact_out_diffimg)
         LOG.interactive("    ***************************************")
         return False
```

Comment:
Do we want to add explicit integration tests for both cases (where sizes don't match, and where sizes DO match but images don't match)?  I'm fine either way, maybe it is just a matter of adding an issue to add unit tests for the output comparisons?

Reply:
I added an Issue:
https://github.com/NRLMMD-GEOIPS/geoips/issues/339